src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -586,6 +600,170 @@ def image_mask_iterator():
+        The nwbfile to add the fluorescence traces to.
+    metadata : dict
+        The metadata for the fluorescence traces.
+    plane_index : int, optional
+        The index of the plane to add the fluorescence traces to.
+
+    Returns
+    -------
+    NWBFile
+        The nwbfile passed as an input with the fluorescence traces added.
+    """
+
+    # Set the defaults and required infrastructure
+    metadata_copy = deepcopy(metadata)
+    default_metadata = get_default_ophys_metadata()
+    metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
+
+    # df/F metadata
+    dff_metadata = metadata_copy["Ophys"]["DfOverF"]
+    dff_name = dff_metadata["name"]
+
+    # Fluorescence traces metadata
+    fluorescence_metadata = metadata_copy["Ophys"]["Fluorescence"]
+    fluorescence_name = fluorescence_metadata["name"]
+
+    # Get traces from the segmentation extractor
+    traces_to_add = segmentation_extractor.get_traces_dict()
+
+    # Filter empty data
+    traces_to_add = {trace_name: trace for trace_name, trace in traces_to_add.items() if trace is not None}
+    # Filter all zero data
+    traces_to_add = {
+        trace_name: trace for trace_name, trace in traces_to_add.items() if any(x != 0 for x in np.ravel(trace))
+    }
+
+    # Early return if there is nothing to add
+    if not traces_to_add:
+        return nwbfile
+
+    # Create a reference for ROIs from the plane segmentation
+    roi_table_region = _create_roi_table_region(
+        segmentation_extractor=segmentation_extractor,
+        nwbfile=nwbfile,
+        metadata=metadata_copy,
+        plane_index=plane_index,
+    )
+
+    rate = (
+        np.nan
+        if segmentation_extractor.get_sampling_frequency() is None
+        else float(segmentation_extractor.get_sampling_frequency())
+    )
+
+    roi_response_series_kwargs = dict(
+        rois=roi_table_region,
+        # TODO: timestamps or rate (for timestamps we need frame_to_time method)
+        # TODO: check for regularity, only use timestamps if irregular timing
+        starting_time=0.0,
+        rate=rate,
+        unit="n.a.",
+    )
+
+    container = defaultdict(
+        lambda: _get_segmentation_data_interface(
+            nwbfile=nwbfile,
+            data_interface_name=fluorescence_name,
+        ),
+        # we expect that the df/F trace name is "Dff"
+        Dff=_get_segmentation_data_interface(
+            nwbfile=nwbfile,
+            data_interface_name=dff_name,
+        ),
+    )
+
+    for trace_name, trace in traces_to_add.items():
+        # Extract the response series metadata
+        trace_name = "RoiResponseSeries" if trace_name == "raw" else trace_name.capitalize()
+        trace_name = trace_name if plane_index == 0 else trace_name + f"_Plane{plane_index}"
+
+        if trace_name in container[trace_name].roi_response_series:
+            continue
+
+        metadata = dff_metadata if trace_name == "Dff" else fluorescence_metadata
```

Comment:
Isn't the trace_name `dff` and not `Dff` which is what you are using for the data interface name?

Reply:
Yeah, I can see why that is confusing. I was using the previous intent of naming traces i.e. raw traces are saved as "RoiResponseSeries" and others are strarting capitalized. Like [here](https://pynwb.readthedocs.io/en/stable/tutorials/domain/ophys.html#sphx-glr-tutorials-domain-ophys-py).